flask/app.py

```diff
@@ -1688,51 +1688,79 @@ def make_response(self, rv):
+                                in the tuple, the status from the tuple is preferred.
+                                Also if a :attr:`response_class` object is used as `response`
+                                and headers are present both in the `response` object and in the tuple,
+                                the headers from both sources are included in the resulting response.
         ======================= ===========================================
 
-        :param rv: the return value from the view function
+        :param retval_from_view_func: the return value from the view function
 
         .. versionchanged:: 0.9
            Previously a tuple was interpreted as the arguments for the
            response object.
         """
-        status_or_headers = headers = None
-        if isinstance(rv, tuple):
-            rv, status_or_headers, headers = rv + (None,) * (3 - len(rv))
-
-        if rv is None:
+        if retval_from_view_func is None:
             raise ValueError('View function did not return a response')
 
-        if isinstance(status_or_headers, (dict, list)):
-            headers, status_or_headers = status_or_headers, None
-
-        if not isinstance(rv, self.response_class):
-            # When we create a response object directly, we let the constructor
-            # set the headers and status.  We do this because there can be
-            # some extra logic involved when creating these objects with
-            # specific values (like default content type selection).
-            if isinstance(rv, (text_type, bytes, bytearray)):
-                rv = self.response_class(rv, headers=headers,
-                                         status=status_or_headers)
-                headers = status_or_headers = None
-            else:
-                rv = self.response_class.force_type(rv, request.environ)
+        if isinstance(retval_from_view_func, self.response_class):
+            return retval_from_view_func
+        elif isinstance(retval_from_view_func, (text_type, bytes, bytearray)):
+            return self.response_class(retval_from_view_func)
+        elif isinstance(retval_from_view_func, tuple):
+            return self._make_response_from_tuple(retval_from_view_func)
+        else:
+            return self.response_class.force_type(retval_from_view_func, request.environ)
 
-        if status_or_headers is not None:
-            if isinstance(status_or_headers, string_types):
-                rv.status = status_or_headers
-            else:
-                rv.status_code = status_or_headers
-        if headers:
-            rv.headers.extend(headers)
+    def _make_response_from_tuple(self, retval_from_view_func):
+        response_body, status, headers = self._extract_response_parts_from_tuple(retval_from_view_func)
 
-        return rv
+        if isinstance(response_body, self.response_class):
+            status, headers  \
```

Comment:
I think the backslash usually comes after the `=` and with only one space before it